flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/disk/DiskCacheManager.java

```diff
@@ -81,6 +91,7 @@ void startSegment(int subpartitionId, int segmentIndex) {
      */
     void append(Buffer buffer, int subpartitionId) {
         subpartitionCacheManagers[subpartitionId].append(buffer);
+        numCachedBytesCounter += buffer.readableBytes();
```

Comment:
Shouldn't we also check whether the max is reached here? Otherwise, we may exceed the max a lot when one of the segments end.